src/app/qgsmaptooladdring.cpp

```diff
@@ -29,6 +29,8 @@ QgsMapToolAddRing::QgsMapToolAddRing( QgsMapCanvas* canvas )
     : QgsMapToolCapture( canvas, QgisApp::instance()->cadDockWidget(), QgsMapToolCapture::CapturePolygon )
 {
   mToolName = tr( "Add ring" );
+  connect( QgisApp::instance(), SIGNAL( newProject() ), this, SLOT( stopCapturing() ) );
+  connect( QgisApp::instance(), SIGNAL( projectRead() ), this, SLOT( stopCapturing() ) );
```

Comment:
> Would it maybe make more sense to connect the layer's destroyed signal instead?

At the moment, a capturing process is not bound to a layer. This PR only targets the issue of orphaned rubberbands after loading a project/creating a new project.
Disallowing to start digitizing on layer A and finish it on layer B is a separate improvement.